packages/strapi-plugin-users-permissions/services/Providers.js

```diff
@@ -64,25 +64,37 @@ exports.connect = (provider, query) => {
           } else {
             strapi.query('user', 'users-permissions').findOne({email: profile.email})
             .then(user => {
```

Comment:
Why are you not using async/await for this function (https://github.com/strapi/strapi/blob/9ddd7f12a80771737f7d5fc387190e6e1d0c38ab/packages/strapi-plugin-users-permissions/controllers/Auth.js#L68)? The code will be more readable and avoid the callback hell.